common/config/genericotlp.go

```diff
@@ -17,23 +23,63 @@ func (g *GenericOTLP) DestType() common.DestinationType {
 }
 
 func (g *GenericOTLP) ModifyConfig(dest ExporterConfigurer, currentConfig *Config) ([]string, error) {
-	url, exists := dest.GetConfig()[genericOtlpUrlKey]
+	config := dest.GetConfig()
+
+	url, exists := config[genericOtlpUrlKey]
 	if !exists {
 		return nil, errors.New("Generic OTLP gRPC endpoint not specified, gateway will not be configured for otlp")
 	}
```

Comment:
nit: we can simplify this error, see utils here: https://github.com/odigos-io/odigos/blob/54b624935cb2398a8761b4c0bdc61f73dcc93506/common/config/utils.go#L145C6-L145C21

```suggestion
	url, exists := config[genericOtlpUrlKey]
	if !exists {
		return nil, errorMissingKey(genericOtlpUrlKey)
	}
```